### PR TITLE
Model loading as a getTinyliciousContainer() replacement

### PR DESCRIPTION
**Pull request by ChumpChief**
getTinyliciousContainer() was always meant to be a temporary utility to help get demos off the ground.  The model loading pattern is not significantly more complex from the app-developer perspective (compare app.ts in particular), while also opening the door to desirable features:

1. Detached container state
2. Plugging in a code loader
    * This immediately enables non-upgrading versioning (i.e. I can safely change my app schema and deploy, but old documents are left behind)
3. Model API surface (rather than IContainer)

This change includes a TinyliciousModelLoader and SessionStorageModelLoader to make the service integration less annoying and a more-1:1 alignment with getTinyliciousContainer.

Opening as a draft to get feedback - I'm not sure I want the ModelLoader to live in example-utils, and I might want to purge getTinyliciousContainer if I'm going to add yet another way to load.

Also see here for more detailed explanation of the container model + model loading pattern:
https://github.com/microsoft/FluidFramework/tree/main/examples/hosts/app-integration/schema-upgrade#container-model

REVIEW REQUEST: I'm curious for your takes on this vs. getTinyliciousContainer in particular.  Do you find it understandable enough as a replacement?  It does complicate the container authoring bit, is it worth that for the advantages listed above?  Should we have some more aqueduct-type support for this pattern before going wider with it?

### CODE REVIEW

**Comment by ChumpChief** [comment]
Latest includes an aqueduct-style container runtime factory to facilitate the pattern.

**Comment by ChumpChief** [comment]
I had a chance to try this pattern further with the SharedWorker driver project, and I feel it's a fairly positive direction.  Since this change would only make changes in the examples (no published packages) I think it's safe to check in and continue to iterate.

File: examples/utils/example-utils/src/modelLoader/interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     */
+    supportsVersion(version: string): Promise<boolean>;
+
+    /**
+     * Create a detached model using the specified version of container code.
+     * Returns an object containing the detached model plus an attach callback.  When invoked, the attach callback
+     * returns a promise that will resolve after attach has completed with the id of the container.
+     * @param version - the container code version to create a model for
+     */
+    createDetached(version: string): Promise<{ model: ModelType; attach: () => Promise<string>; }>;
```
Comment (scottn12) [inline examples/utils/example-utils/src/modelLoader/interfaces.ts:24]: In the schema-upgrade demo I was playing around with an interface for this return value:
```ts
export interface IDetachedModel<ModelType> {
    model: ModelType;
    attach: () => Promise<string>;
}
```
I found it made it easier to store the object for retry scenarios. Let me know what you think!

**Review by scottn12 — APPROVED** [review]
I like this approach! From my experience with it in the schema-upgrade demo it seemed fairly intuitive and easy to work with. I left one suggestion based on my preference, but I could see why you may want to ignore it 😄

File: examples/utils/example-utils/src/modelLoader/interfaces.ts (hunk 1 above)
Comment (ChumpChief) [reply examples/utils/example-utils/src/modelLoader/interfaces.ts]: Makes sense, added in latest!

**Pull request merged**